### PR TITLE
Add external users modifications

### PR DESCRIPTION
**Pull request by jkebertz**
Adding code so that we can handle new optional param to get information on external users by login.

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @jkebertz, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

File: src/main/java/com/box/sdk/BoxUser.java [hunk 1]
```diff
@@ -160,12 +160,60 @@ public static BoxUser getCurrentUser(BoxAPIConnection api) {
      */
     public static Iterable<BoxUser.Info> getAllEnterpriseUsers(final BoxAPIConnection api, final String filterTerm,
             final String... fields) {
+        return getUsersInfoForType(api, filterTerm, null, fields);
+    }
+
+    /**
+     * Gets a limited set of information about an external user.  (A user collaborating
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: The other javadocs use a single space after a period.

File: src/main/java/com/box/sdk/BoxUser.java [hunk 2]
```diff
@@ -160,12 +160,60 @@ public static BoxUser getCurrentUser(BoxAPIConnection api) {
...
+     * match the filter term.  For managed users it matches any users names or emails that
+     * start with the term.  For external, it only does full match on email.  This method
+     * is ideal to use in the case where you have a full email for a user and you don't
+     * know if they're managed or external.
+     * @param  api        the API connection to be used when retrieving the users.
+     * @param filterTerm    The filter term to lookup users by (login for external, login or name for managed)
+     * @param fields        the fields to retrieve. Leave this out for the standard fields.
+     * @return an iterable containing users matching the given email
+     */
+    public static Iterable<BoxUser.Info> getManagedOrExternalUsers(final BoxAPIConnection api,
```
Comment (gcurtis) [inline src/main/java/com/box/sdk/BoxUser.java]: What do you think of calling this `getEnterpriseAndExternalUsers`? I'm wondering if that would be more consistent with the other `getAllEnterpriseUsers` method.

**Comment by gcurtis** [comment]
Thanks! Just a couple of minor comments.

**Comment by gcurtis** [comment]
Squashed and merged in 96c97a947b72945e4f347b5ff3dccab30e81d19f.

**Pull request closed without merging**